### PR TITLE
Finish tests for agents endpoint

### LINKED ISSUE #2835: API: Flask new endpoints Agents
Reimplementation of every agents endpoints implemented on nodejs-based to python for the new API.

Principal roadmap:  https://github.com/wazuh/wazuh/issues/2640

Agent's endpoint to reimplement:
* [x] DELETE /agents
* [x] DELETE /agents/:agent_id
* [x] DELETE /agents/:agent_id/group
* [x] DELETE /agents/:agent_id/group/:group_id
* [x] DELETE /agents/group/:group_id
* [x] DELETE /agents/groups
* [x] DELETE /agents/groups/:group_id
* [x] GET /agents
* [x] GET /agents/:agent_id
* [x] GET /agents/:agent_id/config/:component/:configuration
* [x] GET /agents/:agent_id/group/is_sync
* [x] GET /agents/:agent_id/key
* [x] GET /agents/:agent_id/upgrade_result
* [x] GET /agents/groups
* [x] GET /agents/groups/:group_id
* [x] GET /agents/groups/:group_id/configuration
* [x] GET /agents/groups/:group_id/files
* [x] GET /agents/groups/:group_id/files/:filename
* [x] GET /agents/name/:agent_name
* [x] GET /agents/no_group
* [x] GET /agents/outdated
* [x] GET /agents/stats/distinct
* [x] GET /agents/summary
* [x] GET /agents/summary/os
* [x] POST /agents
* [x] POST /agents/group/:group_id
* [x] POST /agents/groups/:group_id/configuration
* [x] POST /agents/groups/:group_id/files/:file_name
* [x] POST /agents/insert
* [x] POST /agents/restart
* [x] PUT /agents/:agent_id/group/:group_id
* [x] PUT /agents/:agent_id/restart
* [x] PUT /agents/:agent_id/upgrade
* [x] PUT /agents/:agent_id/upgrade_custom
* [x] PUT /agents/:agent_name
* [x] PUT /agents/groups/:group_id
* [x] PUT /agents/restart

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
Hi team,

The tests of the endpoint of agents have been finished for lack of revision.
Corrected all calls that did not return as expected or did not respect the spec.

This PR closes #2835 

Best regards,
Adri

### CODE REVIEW

File: api/test/integration/test_agents_endpoints.tavern.yaml [hunk 1]
```diff
@@ -0,0 +1,5044 @@
...
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+
+    response:
+      status_code: 400
+      body: &error_response
+        code: !anyint
+        dapi_errors:
+          master-node:
```
Comment (druizz90) [inline api/test/integration/test_agents_endpoints.tavern.yaml]: It is possible to use a variable for setting the node name? If you have a master node whose name is different to `master-node`, the test will fail.

File: api/api/controllers/agents_controller.py [hunk 2]
```diff
@@ -233,126 +316,1189 @@ def get_agent_config(agent_id, component, configuration, pretty=False, wait_for_
...
+    """
+    # get body parameters
+    if connexion.request.is_json:
+        agent_list_model = AgentList.from_dict(connexion.request.get_json())
+    else:
+        raise WazuhError(1749, extra_remediation='[official documentation]'
+                                                 '(https://documentation.wazuh.com/current/user-manual/api/reference.html#restart-a-list-of-agents '
+                                                 'to get more information about API call')
+
+    dict = agent_list_model.to_dict()
```
[the remark this answers is not in the document]
Comment (AdriiiPRodri) [reply api/api/controllers/agents_controller.py]: We can't remove this cause the return of AgentInserted.from_dict(connexion.request.get_json()) is a deserialized model, which we must later convert into a dictionary with .to_dict().

File: api/api/controllers/agents_controller.py [hunk 3]
```diff
@@ -233,126 +316,1189 @@ def get_agent_config(agent_id, component, configuration, pretty=False, wait_for_
...
+    # get body parameters
+    if connexion.request.is_json:
+        agent_inserted_model = AgentInserted.from_dict(connexion.request.get_json())
+    else:
+        raise WazuhError(1749, extra_remediation='[official documentation]'
+                                                 '(https://documentation.wazuh.com/current/user-manual/api/reference.html#insert-agent) '
+                                                 'to get more information about API call')
+
+    f_kwargs = {**{}, **agent_inserted_model.to_dict()}
+    try:
```
[the remark this answers is not in the document]
Comment (AdriiiPRodri) [reply api/api/controllers/agents_controller.py]: The model agent_insert had a bug in the function from_dict that has already been fixed, we cannot change this for the same reason that I explained in the conversation below. Here is the fix:
```python
    def from_dict(cls, dikt) -> 'AgentInserted':
        """Returns the dict as a model
        :param dikt: A dict.
        :type: dict
        :return: The Agent of this Agent.
        :rtype: dict
        """
        return util.deserialize_model(dikt, cls)
```

File: api/api/controllers/agents_controller.py [hunk 4]
```diff
@@ -48,13 +57,14 @@ def delete_agents(pretty=False, wait_for_complete=False, ids=None, purge=None, s
...
+    # response = Data(data)
 
     return data, 200
 
-
+@exception_handler
 def get_all_agents(pretty=False, wait_for_complete=False, offset=0, limit=None, select=None, sort=None, search=None,
-                   status=None, q='', older_than=None, os_platform=None, os_version=None, os_name=None, manager=None,
+                   agent_status=None, q='', older_than=None, os_platform=None, os_version=None, os_name=None, manager=None,
                    version=None, group=None, node_name=None, name=None, ip=None):  # noqa: E501
```
[the remark this answers is not in the document]
Comment (AdriiiPRodri) [reply api/api/controllers/agents_controller.py]: Now it' s working.

File: api/test/integration/test_agents_endpoints.tavern.yaml (hunk 1 above)
Comment (JmZero) [reply api/test/integration/test_agents_endpoints.tavern.yaml]: Im going to change the way we check the API error results, now compare only error code

**Pull request merged**